internal/engine/event_watch_manager.go

```diff
@@ -33,18 +34,21 @@ type EventWatchManager struct {
 	uidMap   map[k8s.UID]uidMapEntry
 	uidMapMu sync.RWMutex
 	clock    clockwork.Clock
+	f        feature.Feature
 }
 
-func NewEventWatchManager(kClient k8s.Client, clock clockwork.Clock) *EventWatchManager {
+func NewEventWatchManager(kClient k8s.Client, clock clockwork.Clock, f feature.Feature) *EventWatchManager {
 	return &EventWatchManager{
 		kClient: kClient,
 		uidMap:  make(map[k8s.UID]uidMapEntry),
 		clock:   clock,
+		f:       f,
 	}
 }
 
 func (m *EventWatchManager) needsWatch(st store.RStore) bool {
-	if !k8sEventsFeatureFlagOn() {
+	enabled, err := m.f.IsEnabled("events")
+	if !enabled || err != nil {
 		return false
```

Comment:
FFTI since this is good enough as is, but something I wish I'd thought of on the previous PR:
Ignoring the error here doesn't feel great. Do you think it's worth going with types instead of strings, or is that annoying to do right now and/or likely to scale poorly at scales we're likely to want? e.g.:
```
type Feature interface {
  Enable()
  IsEnabled() bool
  Disable()
}

type FeatureConfig struct {
  Events Feature
}
```

Reply:
Not annoying to do right now and I don't foresee any scaling problems. What problem are you looking to solve? How does that distinguish between features that are off and features that don't exist any better than this?

Reply:
it'd be a compiler error instead of a runtime error

Reply:
errr, it'd be a runtime error on startup, when parsing feature configs, but the code *using* the configs wouldn't have to worry about it

Reply:
Yeah that's what I ended up doing

Reply:
OH now I understand what this is saying. Basically make it a typed map, rather than dynamic.

Reply:
I think I'm going to leave this as is so that we have flexibility in the future to have this backed by something else, like a networked backed store, as much as I like the compile time safety this solution offers.